packages/plugins/typescript/graphql-apollo/src/visitor.ts

```diff
@@ -69,23 +70,26 @@ export class GraphQLApolloVisitor extends ClientSideBaseVisitor<
   }
 
   public get sdkContent(): string {
-    const sdkOperations = this._operationsToInclude.map(x => {
-      const optionalVariables =
-        !x.node.variableDefinitions ||
-        x.node.variableDefinitions.length === 0 ||
-        x.node.variableDefinitions.every(v => v.type.kind !== Kind.NON_NULL_TYPE || v.defaultValue);
-
-      const documentVariableName = x.documentVariableName;
-
-      const operationName = x.node.name.value;
-      return `${camelCase(operationName)}${x.operationType}(variables${optionalVariables ? '?' : ''}: ${
-        x.operationVariablesTypes
-      }) {
-          return client.${GraphQLApolloVisitor.getApolloOperation(x.operationType)}<${
-        x.operationResultType
-      }>({variables, query: ${documentVariableName}})
+    const sdkOperations = this._operationsToInclude.map(
+      ({ node, documentVariableName, operationType, operationResultType, operationVariablesTypes }) => {
+        const optionalVariables =
+          !node.variableDefinitions ||
+          node.variableDefinitions.length === 0 ||
+          node.variableDefinitions.every(v => v.type.kind !== Kind.NON_NULL_TYPE || v.defaultValue);
+
+        const operationName = node.name.value;
+        const operationGenerics = `${operationResultType}, ${operationVariablesTypes}`;
```

Comment:
@MattSchiller I don't know what's going on with apollo, but types are so fucking retarded.

Have a look :smile: 
For mutations:
```typescript
mutate<T = any, TVariables = OperationVariables>(options: MutationOptions<T, TVariables>): Promise<FetchResult<T>>;
```
and for queries:
```
query<T = any, TVariables = OperationVariables>(options: QueryOptions<TVariables, T>): Promise<ApolloQueryResult<T>>;
```
now I've no idea why mutationOptions and queryOptions are flipped

The generator isn't going to look pretty, but I think I can make it work.

Reply:
https://github.com/apollographql/apollo-client/issues/8537 created an issue :smile: